common/djangoapps/student/views.py

```diff
@@ -694,6 +704,42 @@ def _create_recent_enrollment_message(course_enrollment_pairs, course_modes):
         )
 
 
+def _create_credit_availability_message(username, enrolled_courses_dict):
+    """Builds a dict of credit availability for courses.
+
+    Construct a for courses user has completed and has not purchased credit
+    from the credit provider yet.
+
+    Args:
+        username (str): Username of the user.
+        course_enrollment_pairs (list): A list of tuples containing courses, and the associated enrollment information.
+
+    Returns:
+        A dict of courses user is eligible for credit.
+
+    """
+    user_eligibitlites = get_credit_eligibility(username=username)
+    user_purchased_credit = get_purchased_credit_courses(username=username)
+
+    eligibility_messages = {}
+    for course_id, eligibility in user_eligibitlites.iteritems():
+        if course_id not in user_purchased_credit:
+            # TODO: @Will for multiple credit provider how will we pick the duration to display the message
```

Comment:
@wedaly How we can filter the message for multiple credit provider for a given course?